sale_commission_pricelist/models/sale_order.py

```diff
@@ -0,0 +1,25 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Carlos Dauden - Tecnativa <carlos.dauden@tecnativa.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = 'sale.order.line'
+
+    @api.onchange('product_id')
+    def _onchange_product_id_sale_commission_pricelist(self):
+        self.ensure_one()
+        if self.product_id:
```

Comment:
Maybe, there's no `pricelist_id` set in your SO, and thus there's an error due to a `self.ensure_one()` in method `get_product_price_rule`. Thus, change this condition to:
```python
if self.product_id and self.order_id.pricelist_id:
```

Reply:
pricelist_id is required
https://github.com/odoo/odoo/blob/6c044b3675aa58f4bc5b8ce336dfbe98f7c0d243/addons/sale/models/sale.py#L139

Reply:
Well, I'm saying this because our Gitlab CI has failed with that error, so there should be any condition that doesn't fit. I think it's due to calls to `new` instead of `create`. There, required fields are not applied.